Sources/Datadog/DatadogCore/Storage/Writing/FileWriter.swift

```diff
@@ -27,23 +27,18 @@ internal struct FileWriter: Writer {
     ///  - metadata: Encodable metadata to write.
     func write<T: Encodable, M: Encodable>(value: T, metadata: M?) {
         do {
-            let encodedValue = try encode(encodable: value, blockType: .event)
-            let encodedMetadata: Data?
+            var encoded = try encode(encodable: value, blockType: .event)
             if let metadata = metadata {
-                encodedMetadata = try encode(encodable: metadata, blockType: .eventMetadata)
-            } else {
-                encodedMetadata = nil
+                let encodedMetadata = try encode(encodable: metadata, blockType: .eventMetadata)
+                encoded.append(encodedMetadata)
             }
```

Comment:
nit/ According to [the spec](https://datadoghq.atlassian.net/wiki/spaces/RUMP/pages/2982249019/RFC+-+Reducing+the+number+of+view+updates+in+V2#(1)-Add-%E2%80%9Cevent-metadata%E2%80%9D-block-type-in-Core), it should be `metadata` followed by `event` and here we do reverse. We need to either update the code or the spec to be consistent.
 
> It will enable each Feature to optionally store information about the "event" block that it precedes:
>
> `[EVENT 1 METADATA] [EVENT 1] [EVENT 2 METADATA] [EVENT 2] [EVENT 3]`

Reply:
errr, good catch.

Reply:
fixed.